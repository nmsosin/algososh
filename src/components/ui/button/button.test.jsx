import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import {Button} from "./button";

describe('Button component', () => {
  it('render button with text content', () => {
    const buttonWithText = renderer
      .create(<Button text={"there's some text"}/>)
      .toJSON();

    expect(buttonWithText).toMatchSnapshot();
  });

  it('render button without text content', () => {
    const buttonEmpty = renderer
      .create(<Button />)
      .toJSON();

    expect(buttonEmpty).toMatchSnapshot();
  });

  it('render button loader', () => {
    const buttonLoading = renderer
      .create(<Button isLoader={true}/>)
      .toJSON();

    expect(buttonLoading).toMatchSnapshot();
  });

  it('render disabled button', () => {
    const buttonDisabled = renderer
      .create(<Button disabled={true}/>)
      .toJSON();

    expect(buttonDisabled).toMatchSnapshot();
  });

  it('invoke a callback on click', () => {
    const mockCb = jest.fn();

    render(<Button text={'buttonTest'} onClick={mockCb} />)

    const buttonTest = screen.getByText('buttonTest');
    fireEvent.click(buttonTest);

    expect(mockCb).toHaveBeenCalledTimes(1)
  });
});