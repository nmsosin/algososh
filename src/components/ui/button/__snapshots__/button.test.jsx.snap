// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component render button loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button component render button with text content 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    there's some text
  </p>
</button>
`;

exports[`Button component render button without text content 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button component render disabled button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
